core/src/main/java/org/classysheet/core/api/ClassysheetService.java

```
@@ -49,68 +44,16 @@ protected WorkbookData extractWorkbookData(Workbook_ workbook) {
-                    BatchUpdateSpreadsheetRequest requestBody = new BatchUpdateSpreadsheetRequest().setRequests(requests);
-                    sheetService.spreadsheets().batchUpdate(SPREADSHEET_ID, requestBody).execute();
-                }
-
-                SheetMeta sheetMeta = sheetData.sheetMeta();
-                Values sheetValues = sheetService.spreadsheets().values();
-                List<ValueRange> data = new ArrayList<>();
-                ValueRange columnNames = new ValueRange()
-                        .setRange(currentSheetName + "!A1")
-                        .setValues(List.of(
-                                sheetMeta.columnMetas()
-                                        .stream()
-                                        .map(columnMeta -> (Object) columnMeta.name())
-                                        .toList()
-                        ));
-                data.add(columnNames);
-
-                sheetData.streamRows().forEach((row) -> {
-                    data.add(new ValueRange()
-                            .setRange(currentSheetName + "!A" + (row.index() + 2))
-                            .setValues(List.of(row.readValuesAsString())));
-                    System.out.println("Row: " + row);
-                });
-                BatchUpdateValuesRequest valuesBody = new BatchUpdateValuesRequest()
-                        .setValueInputOption("RAW")
-                        .setData(data);
-                sheetValues.batchUpdate(SPREADSHEET_ID, valuesBody).execute();
-            } catch (IOException e) {
-                throw new RuntimeException(e);
-            }
+        if (spreadsheetId == null) {
+            throw new IllegalStateException("spreadsheetId is null. Initialize ClassysheetService with a valid spreadsheetId for Google Sheets.");
         }
+        WorkbookData workbookData = extractWorkbookData(workbook);
+        assert googleSpreadsheetConnector != null;
```

(nitpick) We avoid asserts this way, because most JVMs run with assert checks off.
And we still want to "fail fast" in such cases.

Let's use `Objects.assertNotNull(...)`instead.